### PR TITLE
[SPARK-6765] Fix test code style for SQL

### PR DESCRIPTION
**Pull request by rxin**
So we can turn style checker on for test code.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 1]
```diff
@@ -202,7 +204,7 @@ class AnalysisSuite extends FunSuite with BeforeAndAfter {
 
   case class UnresolvedTestPlan() extends LeafNode {
     override lazy val resolved = false
```
Comment (liancheng) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala]: Do we also want public fields have explicit type too?

Comment (rxin) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala]: Yes - but there is no rule for it yet.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala [hunk 2]
```diff
@@ -45,12 +45,13 @@ class PlanTest extends FunSuite {
...
     val normalized1 = normalizeExprIds(plan1)
     val normalized2 = normalizeExprIds(plan2)
-    if (normalized1 != normalized2)
+    if (normalized1 != normalized2) {
       fail(
         s"""
           |== FAIL: Plans do not match ===
           |${sideBySide(normalized1.treeString, normalized2.treeString).mkString("\n")}
         """.stripMargin)
+      }
```
Comment (liancheng) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala]: Indentation is off here.

File: sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ class QueryTest extends PlanTest {
     checkAnswer(df, Seq(expectedAnswer))
   }
 
-  def sqlTest(sqlString: String, expectedAnswer: Seq[Row])(implicit sqlContext: SQLContext): Unit = {
+  def sqlTest(sqlString: String, expectedAnswer: Seq[Row])(implicit sqlContext: SQLContext) {
```
Comment (liancheng) [inline sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala]: Why removing return type of this one?

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 4]
```diff
@@ -182,7 +182,10 @@ class SQLQuerySuite extends QueryTest with BeforeAndAfterAll {
       Row(java.sql.Timestamp.valueOf("1969-12-31 16:00:00.002")))
 
     checkAnswer(sql(
-      "SELECT time FROM timestamps WHERE time IN ('1969-12-31 16:00:00.001','1969-12-31 16:00:00.002')"),
+      """"
```
Comment (liancheng) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: 4 double quotes?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala [hunk 5]
```diff
@@ -115,11 +115,36 @@ class InsertIntoHiveTableSuite extends QueryTest with BeforeAndAfter {
...
-    sql("INSERT OVERWRITE TABLE table_with_partition  partition (p1='a',p2='b',p3='c',p4='c',p5='1') SELECT 'blarr' FROM tmp_table")
-    sql("INSERT OVERWRITE TABLE table_with_partition  partition (p1='a',p2='b',p3='c',p4='c',p5='2') SELECT 'blarr' FROM tmp_table")
-    sql("INSERT OVERWRITE TABLE table_with_partition  partition (p1='a',p2='b',p3='c',p4='c',p5='3') SELECT 'blarr' FROM tmp_table")
-    sql("INSERT OVERWRITE TABLE table_with_partition  partition (p1='a',p2='b',p3='c',p4='c',p5='4') SELECT 'blarr' FROM tmp_table")
+    sql(
+      s"""
+         |CREATE TABLE table_with_partition(c1 string)
+         |PARTITIONED by (p1 string,p2 string,p3 string,p4 string,p5 string)
+         |location '${tmpDir.toURI.toString}'
+        """.stripMargin)
```
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala]: One less space.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala [hunk 6]
```diff
@@ -887,7 +886,11 @@ abstract class ParquetPartitioningTest extends QueryTest with BeforeAndAfterAll
 
     test(s"SPARK-5775 read struct from $table") {
       checkAnswer(
-        sql(s"SELECT p, structField.intStructField, structField.stringStructField FROM $table WHERE p = 1"),
+        sql(
+          s"""
+             |SELECT p, structField.intStructField, structField.stringStructField
+             |FROM $table WHERE p = 1
+            """.stripMargin),
```
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala]: One less space before the quotes.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala [hunk 7]
```diff
@@ -237,7 +238,11 @@ class HiveQuerySuite extends HiveComparisonTest with BeforeAndAfter {
   }
 
   createQueryTest("modulus",
-    "SELECT 11 % 10, IF((101.1 % 100.0) BETWEEN 1.01 AND 1.11, \"true\", \"false\"), (101 / 2) % 10 FROM src LIMIT 1")
+    """
+      |SELECT 11 % 10,
+      |       IF((101.1 % 100.0) BETWEEN 1.01 AND 1.11, "true", "false"), (101 / 2) % 10
+      |FROM src LIMIT 1
+    """.stripMargin)
```
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala]: Golden answers must also be regenerated as MD5 of the query string is used to name the golden answer file. This also applies to the other 2 query string changes in this file.

File: sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala (hunk 3 above)
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala]: it was too long - either wrap it, or just remove :Unit

Comment (marmbrus) [inline sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala]: I believe this form of method is going to be deprecated.

**Comment by marmbrus** [comment]
For @rxin 's sanity I'll merge this now :)

We can update the final minor comments in followups.

**Comment by marmbrus** [comment]
Actually, apache git is down :(

**Comment by rxin** [comment]
Ok I've merged it.

**Pull request closed without merging**